src/contexts/Auth.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+interface AuthContextProps {
+  user: {
+    username: string
+    role: string
+  } | null;
+  setUser: (user: { username: string, role: string } | null) => void;
+}
+
+const initialAuthContext: AuthContextProps = {
+  user: null,
+  setUser: () => { },
+};
+
+export const AuthContext = createContext<AuthContextProps>(initialAuthContext);
+
+export const useAuthContext = () => useContext(AuthContext)
```

Comment:
lets keep it consistent and move this one into the hooks folder with the other contexts, but lets remove `Context` inside the name. Let it be only `useAuth`

Reply:
Fixed.